src/login/LoginPage.jsx

```diff
@@ -1,435 +1,266 @@
+      setSurveyCookie('login');
     }
-  }
+  }, [loginResult]);
 
-  getEnterPriseLoginURL() {
-    return getConfig().LMS_BASE_URL + ENTERPRISE_LOGIN_URL;
-  }
+  const validateFormFields = (payload) => {
+    const { emailOrUsername, password } = payload;
+    const fieldErrors = { ...errors };
 
-  handleSubmit = (e) => {
-    e.preventDefault();
-    if (this.props.resetPassword) {
-      this.props.loginRemovePasswordResetBanner();
+    if (emailOrUsername === '') {
+      fieldErrors.emailOrUsername = formatMessage(messages['email.validation.message']);
+    } else if (emailOrUsername < 3) {
+      fieldErrors.emailOrUsername = formatMessage(messages['username.or.email.format.validation.less.chars.message']);
+    }
+    if (password === '') {
+      fieldErrors.password = formatMessage(messages['password.validation.message']);
     }
-    this.setState({ isSubmitted: true });
-    const { emailOrUsername, password } = this.state;
-    const emailValidationError = this.validateEmail(emailOrUsername);
-    const passwordValidationError = this.validatePassword(password);
 
-    if (emailValidationError !== '' || passwordValidationError !== '') {
-      this.props.setLoginFormData({
-        errors: {
-          emailOrUsername: emailValidationError,
-          password: passwordValidationError,
-        },
-      });
-      this.props.loginRequestFailure({
-        errorCode: INVALID_FORM,
-      });
+    return { ...fieldErrors };
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (showResetPasswordSuccessBanner) {
+      props.dismissPasswordResetBanner();
+    }
+
+    const formData = { ...formFields };
+    const validationErrors = validateFormFields(formData);
+    if (validationErrors.emailOrUsername || validationErrors.password) {
+      setErrors({ ...validationErrors });
+      setErrorCode(prevState => ({ type: INVALID_FORM, count: prevState.count + 1 }));
       return;
     }
 
-    const payload = {
-      email_or_username: emailOrUsername, password, ...this.queryParams,
-    };
-    this.props.loginRequest(payload);
+    // add query params to the payload
+    const payload = { email_or_username: formData.emailOrUsername, password: formData.password, ...queryParams };
+    props.loginRequest(payload);
   };
 
-  handleOnFocus = (e) => {
-    const { errors } = this.state;
-    errors[e.target.name] = '';
-    this.props.setLoginFormData({
-      errors,
-    });
+  const handleOnChange = (event) => {
+    const { name, value } = event.target;
+    setFormFields(prevState => ({ ...prevState, [name]: value }));
   };
 
-  handleOnBlur = (e) => {
-    const payload = {
-      [e.target.name]: e.target.value,
-    };
-    this.props.setLoginFormData(payload);
+  const handleOnFocus = (event) => {
+    const { name } = event.target;
+    setErrors(prevErrors => ({ ...prevErrors, [name]: '' }));
   };
 
-  handleForgotPasswordLinkClickEvent = () => {
+  const handleForgotPasswordLinkClick = () => {
```

Comment:
if we are only sending a track event here, can we rename it to `trackForgotPasswordLinkClick`?